### PR TITLE
Do not allow put mapping on follower

### PR DESCRIPTION
**Pull request by dnhatn**
Today, the mapping on the follower is managed and replicated from its
leader index by the ShardFollowTask. Thus, we should prevent users 
from modifying the mapping on the follower indices.

Relates #30086

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/CcrRequests.java [hunk 1]
```diff
@@ -24,8 +32,23 @@ public static ClusterStateRequest metaDataRequest(String leaderIndex) {
...
     }
+
+    public static final MappingRequestOriginValidator CCR_PUT_MAPPING_REQUEST_VALIDATOR = (request, state, indices) -> {
+        final List<Index> followingIndices = Arrays.stream(indices)
+            .filter(index -> {
+                final IndexMetaData indexMetaData = state.metaData().index(index);
+                return indexMetaData != null && CcrSettings.CCR_FOLLOWING_INDEX_SETTING.get(indexMetaData.getSettings());
+            }).collect(Collectors.toList());
+        if (followingIndices.isEmpty() == false && "ccr".equals(request.origin()) == false) {
+            return new IllegalStateException("can't put mapping to the following indices [" +
```
Comment (martijnvg) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/CcrRequests.java]: Maybe throw an ElasticSearchStatus exception here with a 403 response code? `IllegalStateException` will result into a 500 response code and in this case a user is something that is not allowed.

**Review by martijnvg — APPROVED** [review]
Left one comment, but LGTM. When I thought about this, I also saw no other way to implement this without introducing an extension point.

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/CcrRequests.java (hunk 1 above)
Comment (dnhatn) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/CcrRequests.java]: ++. I pushed https://github.com/elastic/elasticsearch/pull/37675/commits/e83679dae5415f575c6dce8e0e02995ce599ef29.

**Review by jasontedor — CHANGES_REQUESTED** [review]
The change looks good but I have one comment about the naming. We use "origin" as a concept in the security code base and I am concerned about the "origin" here being confused with the "origin" there. Since we are handing the entire mapping request to the validator (and not really any notion of "origin" or "source") I would be inclined to change the name to remove "origin" and simply say that they are put mapping request validators. What do you think?

**Comment by dnhatn** [comment]
> We use "origin" as a concept in the security code base and I am concerned about the "origin" here being confused with the "origin" there. Since we are handing the entire mapping request to the validator (and not really any notion of "origin" or "source") I would be inclined to change the name to remove "origin" and simply say that they are put mapping request validators. 

+1. I pushed https://github.com/elastic/elasticsearch/pull/37675/commits/d2a61d52a8214f4b348dd1694aa1087d05f88437 to remove the origin from the validator and documentation. @jasontedor Could you please have another look? Thank you!

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by dnhatn** [comment]
Thanks @martijnvg and @jasontedor for reviewing.

**Pull request merged**